### PR TITLE
Added LU and triangular solvers Solvers

### PR DESCRIPTION
**Pull request by pavanky**

### CODE REVIEW

File: src/api/c/lu.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ af_err af_lu(af_array *lower, af_array *upper, af_array *pivot, const af_array i
     return AF_SUCCESS;
 }
 
-af_err af_lu_inplace(af_array *pivot, af_array in)
+af_err af_lu_inplace(af_array *pivot, const af_array in, const bool is_lapack_piv)
```
Comment (shehzan10) [inline src/api/c/lu.cpp]: af_array in should not be const

File: test/solve_dense.cpp [hunk 2]
```diff
@@ -53,26 +53,114 @@ void solveTester(const int m, const int n, const int k, double eps)
...
+    {                                                   \
+        solveTester<T>(800, 1000, 200, eps);            \
+    }                                                   \
+    TEST(SOLVE, T##RectUnderMultiple)                   \
+    {                                                   \
+        solveTester<T>(1536, 2048, 400, eps);           \
+    }                                                   \
+    TEST(SOLVE, T##RectOverMultiple)                    \
+    {                                                   \
+        solveTester<T>(1536, 1024, 1, eps);             \
```
Comment (umar456) [inline test/solve_dense.cpp]: These tests are too large. make smaller tests. I will try to have a different target for large tests.

File: src/api/c/lu.cpp (hunk 1 above)
Comment (pavanky) [inline src/api/c/lu.cpp]: Fixed

**Comment by pavanky** [comment]
Triangle solve is failing on nvidia gpus. Fixing this.

**Comment by pavanky** [comment]
Fixed. @umar456 merge when ready.

**Pull request merged**